contracts/governance/TrueFiVault.sol

```diff
@@ -0,0 +1,158 @@
+ * funds in vault.
+ */
+contract TrueFiVault {
+    using SafeMath for uint256;
+
+    address public owner;
+    address public beneficiary;
+    uint256 public expiry;
+
+    GovernorAlpha public governance;
+    IERC20 public tru;
+    StkTruToken public stkTru;
+    ITrueRatingAgencyV2 public ratingAgency;
+
+    event WithdrawTo(address recipient);
+
+    constructor(
+        address _beneficiary,
+        uint256 _amount,
+        uint256 _duration,
+        GovernorAlpha _governance,
+        ITrueRatingAgencyV2 _ratingAgency
+    ) public {
+        owner = msg.sender;
+        beneficiary = _beneficiary;
+        expiry = block.timestamp.add(_duration);
+
+        governance = _governance;
+        tru = IERC20(address(governance.trustToken()));
+        stkTru = StkTruToken(address(governance.stkTRU()));
+        ratingAgency = _ratingAgency;
+
+        require(tru.transferFrom(owner, address(this), _amount), "TrueFiVault: insufficient balance.");
+    }
+
+    /**
+     * @dev Throws if called by any account other than the beneficiary.
+     */
+    modifier onlyBeneficiary() {
+        require(msg.sender == beneficiary, "only beneficiary");
+        _;
+    }
+
+    /**
+     * @dev Throws if called by any account other than the owner.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, "only owner");
+        _;
+    }
+
+    /**
+     * @dev Allow owner to withdraw funds in case of emergency or mistake
+     */
+    function withdrawToOwner() external onlyOwner {
+        _withdrawTo(owner);
+    }
+
+    /**
+     * @dev Withdraw funds to beneficiary after expiry time
+     */
+    function withdrawToBeneficiary() external onlyBeneficiary {
+        require(block.timestamp >= expiry, "TrueFiVault: beneficiary cannot withdraw before expiration");
+        _withdrawTo(beneficiary);
+    }
+
+    /**
+     * @dev Internal function to withdraw funds to recipient
+     */
+    function _withdrawTo(address recipient) private {
+        emit WithdrawTo(recipient);
+        require(tru.transfer(recipient, tru.balanceOf(address(this))), "TrueFiVault: insufficient balance.");
+        require(stkTru.transfer(recipient, stkTru.balanceOf(address(this))), "TrueFiVault: insufficient balance.");
+    }
+
+    /**
+     * @dev Cast vote in governance for `proposalId`
+     * Uses both TRU and stkTRU balance
+     * @param proposalId Proposal ID
+     * @param support Vote boolean
+     */
+    function castVote(uint256 proposalId, bool support) external onlyBeneficiary {
+        governance.castVote(proposalId, support);
+    }
+
+    /**
+     * @dev Rate YES on a loan by staking TRU
+     * @param id Loan ID
+     */
+    function rateLoanYes(address id) external onlyBeneficiary {
+        ratingAgency.yes(id);
+    }
+
+    /**
+     * @dev Rate NO on a loan by staking TRU
+     * @param id Loan ID
+     */
+    function rateLoanNo(address id) external onlyBeneficiary {
+        ratingAgency.no(id);
+    }
+
```

Comment:
Add one more function here to remove rating (i.e. vote 0 yes and 0 no)

Reply:
ITrueRatingAgencyV2 doesn't have that function visible, so I added it to the interface in PR #610. Will update it here as well once that's checked in.

Reply:
done. this looks fine now as is